presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -494,4 +498,52 @@ else if (parsePos < 0) {
             return ~bestInvalidPos;
         }
     }
+
+    public static SqlTimestamp sqlTimestampOf(int year,
+            int monthOfYear,
+            int dayOfMonth,
+            int hourOfDay,
+            int minuteOfHour,
+            int secondOfMinute,
+            int millisOfSecond,
+            DateTimeZone baseZone,
+            TimeZoneKey timestampZone,
+            ConnectorSession session)
+    {
+        if (session.isLegacyTimestamp()) {
```

Comment:
Not really reviewing this yet, but...

"Introduce sqlTime(stamp)Of method to avoid direct constructor calls" cmt msg suggests a refactor and here you seem to add logic